### PR TITLE
Adding NOAA-20, NOAA-21, S-NPP AWS downloads

### PR DESCRIPTION
**Pull request by mindyls**
Required to fix GEOIPS/geoips#666

This is required to download NOAA VIIRS test datasets for the data fusion tutorial.

```
Package: geoips_full
Total run time: 2017 seconds
Number data types run: 56
Number data types failed: 0
Wed Aug  7 21:34:26 UTC 2024
```

### CODE REVIEW

**Review by christopher-selman — APPROVED** [review]

File: tests/download_noaa_aws.sh [hunk 1]
```diff
@@ -87,7 +104,6 @@ elif [[ "$satellite" == "geokompsat" ]]; then
     echo "rclone --config $rclone_conf lsf $rclone_path"
     files=`rclone --config $rclone_conf lsf $rclone_path`
     echo "COMPARE: ${yyyy}${mm}${dd}${hh}${mn}"
-    break
```
Comment (mindyls) [inline tests/download_noaa_aws.sh:90]: I don't think this was supposed to be here under the geokompsat section

Comment (cpcamacho) [reply tests/download_noaa_aws.sh:90]: I do not think so either! Resolving this conversation

**Review by evrose54 — APPROVED** [review]
LGTM; not sure if this is showing up for any of you but we might want to look at this before merging. Screenshot 2024-08-07 at 3 43 17 PM

File: tests/download_noaa_aws.sh [hunk 2]
```diff
@@ -25,13 +25,30 @@ if [[ "$8" == "" && -z "$GEOIPS_PACKAGES_DIR" ]]; then
...
     echo "       \$GEOIPS_TESTDATA_DIR/test_data_noaa_aws/data/<satellite>/<YYYYmmdd>/<HHMN>"
-    echo "    rclone_conf: if 'default' or not specified, defaults to:"
+    echo "    rclone_conf: "
+    echo "        if 'default' or not specified, defaults to:"
     echo "        \$GEOIPS_PACKAGES_DIR/geoips/setup/rclone_setup/rclone.conf"
-    echo "    wildcard_list: list of strings to match in filenames (ie, channels). Defaults to all files for dtg"
+    echo "    collection:"
+    echo "        viirs"
+    echo "           VIIRS-IMG-GEO-TC"
+    echo "           VIIRS-I5-SDR"
```
Comment (mindyls) [inline tests/download_noaa_aws.sh:48]: How do we differentiate between "collection' and "wildcard_list" ?  Is collection used for all of the satellites?

"collection' is the subdirectory, and "wildcard_list" are substrings from the filenames?  Is this always true?

**Review by AJThorpe — APPROVED** [review]

**Review by christopher-selman — APPROVED** [review]

File: tests/download_noaa_aws.sh [hunk 3]
```diff
@@ -25,13 +25,30 @@ if [[ "$8" == "" && -z "$GEOIPS_PACKAGES_DIR" ]]; then
...
+    echo "        himawari8"
+    echo "        himawari9"
+    echo "        geokompsat"
+    echo "        noaa-20"
+    echo "        noaa-21"
+    echo "        snpp"
+    echo "        jpss"
+    echo "    testdata_dir: "
+    echo "        if 'default' or not specified, defaults to: "
     echo "       \$GEOIPS_TESTDATA_DIR/test_data_noaa_aws/data/<satellite>/<YYYYmmdd>/<HHMN>"
```
Comment (jsolbrig) [inline tests/download_noaa_aws.sh:41]: I wonder if, in the future, this should default to `test_data_noaa_<satellite_name>`.

**Review by jsolbrig — DISMISSED** [review]

File: tests/download_noaa_aws.sh (hunk 2 above)
Comment (cpcamacho) [reply tests/download_noaa_aws.sh:48]: I'm not entirely sure if "collection" is the right terminology, but yes, that defines the subdirectory under the S3 bucket. "wildcard_list" only applies to the substrings in the filenames. I think that will always be true, but I could be wrong.

File: tests/download_noaa_aws.sh (hunk 3 above)
Comment (mindyls) [reply tests/download_noaa_aws.sh:41]: We'll consider that whne we switch to Evan's python downloader.

**Review by cpcamacho — APPROVED** [review]

**Review by jsolbrig — APPROVED** [review]

**Pull request merged**